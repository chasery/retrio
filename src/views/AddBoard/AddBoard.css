.AddBoard__wrapper {
  display: flex;
  justify-content: center;
  padding: 1rem 1.5rem;
}

@media all and (min-width: 50rem) {
  .AddBoard__wrapper {
    max-width: 64rem;
    align-self: center;
    margin: 0 auto;
  }
}
