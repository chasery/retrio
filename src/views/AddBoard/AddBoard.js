import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import Header from '../../components/Header/Header';
import Form from '../../components/Form/Form';
import FormField from '../../components/FormField/FormField';
import Error from '../../components/Error/Error';
import './AddBoard.css';

function AddBoard(props) {
  const history = useHistory();
  const [team, setTeam] = useState('');
  const [boardName, setBoardName] = useState('');
  const [error, setError] = useState(null);

  // Response object I am expecting for teams
  const teams = [
    {
      id: 1,
      name: 'Deploy Happy',
    },
    {
      id: 2,
      name: 'Someone Else Broke It',
    },
    {
      id: 3,
      name: 'Team Name Pending',
    },
  ];

  const handleAddBoard = (e) => {
    e.preventDefault();
  };

  const renderTeamOptions = (teams) => {
    return teams.map((team) => <option value={team.id}>{team.name}</option>);
  };

  return (
    <>
      <Header />
      <main role='main'>
        <section className='AddBoard'>
          <div className='AddBoard__wrapper'>
            <Form id='AddBoard' onSubmit={(e) => handleAddBoard(e)}>
              <div className='Form__header'>
                <h2>Add Board</h2>
              </div>
              <div className='Form__body'>
                <p>Add a Retrio board to begin collaborating with your team.</p>
                <div className='FormField'>
                  <label className='FormField__label' htmlFor={team}>
                    Team<span className='FormField__required'>*</span>
                  </label>
                  <div class='FormField__select'>
                    <select
                      id='team'
                      required
                      value={team}
                      onChange={(e) => setTeam(e.target.value)}
                    >
                      <option value=''>Select Team</option>
                      {renderTeamOptions(teams)}
                    </select>
                  </div>
                </div>
                <FormField
                  id='boardName'
                  label='Board Name'
                  type='text'
                  isRequired={true}
                  onChange={(e) => setBoardName(e.target.value)}
                  value={boardName}
                />
                {error ? <Error message={error} /> : null}
                <div className='Form__controls'>
                  <button
                    className='Form__button secondary'
                    type='button'
                    onClick={() => history.push(`/boards`)}
                  >
                    Cancel
                  </button>
                  <button className='Form__button primary' type='submit'>
                    Add Board
                  </button>
                </div>
              </div>
            </Form>
          </div>
        </section>
      </main>
    </>
  );
}

export default AddBoard;
