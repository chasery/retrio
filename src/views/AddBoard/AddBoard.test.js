import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddBoard from './AddBoard';

describe('AddBoard component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<AddBoard />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
