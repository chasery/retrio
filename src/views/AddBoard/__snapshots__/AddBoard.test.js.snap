// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBoard component renders without error 1`] = `
<Fragment>
  <Header />
  <main
    role="main"
  >
    <section
      className="AddBoard"
    >
      <div
        className="AddBoard__wrapper"
      >
        <Form
          id="AddBoard"
          onSubmit={[Function]}
        >
          <div
            className="Form__header"
          >
            <h2>
              Add Board
            </h2>
          </div>
          <div
            className="Form__body"
          >
            <p>
              Add a Retrio board to begin collaborating with your team.
            </p>
            <div
              className="FormField"
            >
              <label
                className="FormField__label"
                htmlFor=""
              >
                Team
                <span
                  className="FormField__required"
                >
                  *
                </span>
              </label>
              <div
                class="FormField__select"
              >
                <select
                  id="team"
                  onChange={[Function]}
                  required={true}
                  value=""
                >
                  <option
                    value=""
                  >
                    Select Team
                  </option>
                  <option
                    key="1"
                    value={1}
                  >
                    Deploy Happy
                  </option>
                  <option
                    key="2"
                    value={2}
                  >
                    Someone Else Broke It
                  </option>
                  <option
                    key="3"
                    value={3}
                  >
                    Team Name Pending
                  </option>
                </select>
              </div>
            </div>
            <FormField
              id="boardName"
              isRequired={true}
              label="Board Name"
              onChange={[Function]}
              type="text"
              value=""
            />
            <div
              className="Form__controls"
            >
              <button
                className="Form__button secondary"
                onClick={[Function]}
                type="button"
              >
                Cancel
              </button>
              <button
                className="Form__button primary"
                type="submit"
              >
                Add Board
              </button>
            </div>
          </div>
        </Form>
      </div>
    </section>
  </main>
</Fragment>
`;
